MapboxDirections/Match/MBMatchOptions.swift

```diff
@@ -0,0 +1,86 @@
+import Foundation
+
+@objc(MBMatchOptions)
+open class MatchOptions: RouteOptions {
+    public init(locations: [CLLocation], profileIdentifier: MBDirectionsProfileIdentifier? = nil) {
+        let waypoints = locations.map {
+            Waypoint(location: $0)
+        }
+        self.timestamps = locations.map { $0.timestamp }
+        super.init(waypoints: waypoints, profileIdentifier: profileIdentifier)
+    }
+    
+    @objc open var resample: Bool = false
+    
+    @objc open var timestamps: [Date]?
+    
+    public required init?(coder decoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override internal var params: [URLQueryItem] {
+        var params: [URLQueryItem] = [
+            URLQueryItem(name: "geometries", value: String(describing: shapeFormat)),
+            URLQueryItem(name: "overview", value: String(describing: routeShapeResolution)),
+            URLQueryItem(name: "steps", value: String(includesSteps)),
+            URLQueryItem(name: "language", value: locale.identifier),
```

Comment:
Most of these parameters are copied from the superclass. Instead of repeating the code here, get the superclass’s `params` and add to them in this implementation.